sensor/common/registry/registry_store.go

```diff
@@ -170,9 +183,14 @@ func genIntegrationName(prefix string, namespace string, registry string) string
 
 // UpsertRegistry upserts the given registry with the given credentials in the given namespace into the store.
 func (rs *Store) UpsertRegistry(ctx context.Context, namespace, registry string, dce config.DockerConfigEntry) error {
-	secure, err := rs.checkTLS(ctx, registry)
+	secure, skip, err := rs.checkTLS(ctx, registry)
 	if err != nil {
-		return errors.Wrapf(err, "unable to check TLS for registry %q", registry)
+		return err
+	}
+
+	if skip {
+		log.Debugf("Skipping upsert for %q from %q due to previous TLS check error", registry, namespace)
+		return nil
```

Comment:
I'm thinking maybe we should return the error. I know clients of this function just log the error, so it doesn't matter much at this point, but I think this is technically an error and should be returned as such to the client.

Perhaps we can update `UpsertRegistry` to return a `bool` as well to indicate if it was actually upserted or not? Or just the error and we deal with the repeated log messages.

What do you think?